webrender/src/renderer.rs

```diff
@@ -3223,12 +3229,18 @@ impl Renderer {
 
             let mut program = self.cs_text_run.get(&mut self.device).unwrap();
             program.bind_locals(&self.device.device, projection, 0);
-            for (_texture_id, instances) in &target.alpha_batcher.text_run_cache_prims {
-                // TODO: bind_texture: BatchTextures::color(*texture_id)
-                    program.bind_instances(
-                        &self.device.device,
-                        &instances.iter().map(|pi| pi.into()).collect::<Vec<PrimitiveInstance>>(),
+            for (texture_id, instances) in &target.alpha_batcher.text_run_cache_prims {
+                for i in 0 .. BatchTextures::color(*texture_id).colors.len() {
+                    self.texture_resolver.bind(
+                        &BatchTextures::color(*texture_id).colors[i],
+                        TextureSampler::color(i),
+                        &mut self.device,
                     );
+                }
+                program.bind_instances(
+                    &self.device.device,
+                    &instances.iter().map(|pi| pi.into()).collect::<Vec<PrimitiveInstance>>(),
```

Comment:
can we have `bind_instances` accepting an iterator to avoid collections?

Reply:
I just fixed the indentation here, but sure. I made a note for myself (actually for @zakorgy :) )  to look into this.